pkg/clusterversion/cv.go

```diff
@@ -111,26 +132,10 @@ func (c *clusterVersionClient) EnsureDesiredConfig(uc *upgradev1alpha1.UpgradeCo
 		if err != nil {
 			return false, err
 		}
+		return true, nil
 	}
 
-	// Commence upgrade using Image
-	if !empty(desired.Image) && !empty(desired.Version) && empty(desired.Channel) {
-		if clusterVersion.Spec.DesiredUpdate == nil || (clusterVersion.Spec.DesiredUpdate.Image != desired.Image) {
-			logger.Info(fmt.Sprintf("Setting ClusterVersion to Image %s", desired.Image))
-			clusterVersion.Spec.DesiredUpdate = &configv1.Update{Image: desired.Image}
-			err = c.client.Update(context.TODO(), clusterVersion)
-			if err != nil {
-				return false, err
-			}
-		}
-	}
-
-	// If neither (version+channel) nor (image) is defined, throw error.
-	if (empty(desired.Version) && empty(desired.Channel)) && empty(desired.Image) {
-		return false, fmt.Errorf("need either (version+channel) or (image) defined in UpgradeConfig")
-	}
-
-	return true, nil
+	return false, nil
```

Comment:
This occurs when a non valid upgradesource is present? we should throw error here?

Reply:
updated to print error message when it returns false, also fixed several other places with the same `false, nil` condition